vorbisgain/0001-temp_files.patch

```diff
@@ -0,0 +1,75 @@
+Description: Apply patch to use temp files, which are dependent
+ on the file which is beeing processed, instead always using the same
+ filename, which can result to data loss in scenarios where two or more
+ vorbis processes are running parallel.
+Author: Pavel N. Krivitsky <pavel@krivitsky.name>
+Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=505164
+Forwarded: no
+---
+ vorbis.c |   17 +++++++++--------
+ 1 file changed, 9 insertions(+), 8 deletions(-)
+
+--- a/vorbis.c
++++ b/vorbis.c
+@@ -56,7 +56,7 @@
+ #define PROGRESS_FORMAT_SIZE 8
+ #define MIN_FILENAME_SIZE 5
+ #define MIN_MIDDLE_TRUNCATE_SIZE 20
+-#define TEMP_NAME "vorbisgain.tmpXXXXXX"
++#define TEMP_EXT ".vgain.tmpXXXXXX"
+ 
+ 
+ /**
+@@ -684,11 +684,10 @@
+         }
+     }
+ 
+-    /* Make sure temp is in same folder as file. And yes, the malloc is larger
+-     * than necessary (and not always needed). Lets keep it simple though (at
+-     * the expense of a few bytes)...
++    /* Construct a temporary file name by appending TEMP_EXT to
++     * the name of the file being modified.
+      */
+-    temp_name = malloc(strlen(filename) + sizeof(TEMP_NAME));
++    temp_name = malloc(strlen(filename)*sizeof(char) + sizeof(TEMP_EXT));
```

Comment:
I'm pretty sure this patch isn't even correct. It needs `+ 1` for trailing `\0`.

Reply:
Oh, nevermind its actually correct as `TEMP_EXT` includes it.

I'd still call this low quality but oh well.